src/database/db_connection.py

```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Ruchita Nathani <ruchita.nathani@fau.de>
+
+import pymongo
+
+_client = None
+
+
+def mongo_connection(collection_name="default"):
+    global _client
+    if _client is None:
+        _client = pymongo.MongoClient("mongodb://root:example@mongodb:27017/")
+    db = _client["leads_enriched"]
+    collection = db[collection_name]
+    return collection
```

Comment:
For some clarification:
So what i mean is that we might want a wrapper class that handles all interaction with the DB. For example let’s call it MonogDBHandler. Then we could implement functions like
MongoDBHandler.insert_lead(lead)
Which takes a Lead object and takes care of the logic to make it into a dict accepted by the pymongo client. Same for the retrieval by an id or searching leads. This is because i think that pymongo expects dicts as input and not any type of object.
This also allows us to abstract away from the pymongo implementation and call more general functions.
For the other thing with the csv replacement etc i wasn’t aware of the scope of your task. But for the future we can probably make different collections for base leads and enriched leads. But let’s brainstorm on that in future sprints. If the whole Handler class is also out of scope for your task then we can also do that in the next sprints

Reply:
Yes, Felix I completely agree with your suggestion. It would be great if we could do that in the next sprint.